core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -727,6 +798,7 @@ object SnappyStoreHiveCatalog {
 
   private[this] var relationDestroyVersion = 0
   private val relationDestroyLock = new ReentrantReadWriteLock()
+  private val alterTableLock = new Object
```

Comment:
Since they will create a column table, the GemXD layer will ultimately fail. Note that this can happen across clients too e.g. via embedded mode vs in split cluster mode which this lock cannot prevent. So what we do need to ensure is that if the index table creation in GemXD does fail with any exception, then the catalog artifacts should be cleaned properly.